### PR TITLE
Execute block

### PR DESCRIPTION
**Pull request by Daniel-K-Ivanov**
- Adds implementation of the 4 types of DigestItems
- Adds the DigestItems support in the Header
- Enhances the Block implementation
- Fixes the Extrinsics encoding / decoding (there was a missing bool property

### CODE REVIEW

File: runtime/assembly/__tests__/mock-builder.ts [hunk 1]
```diff
@@ -12,85 +14,143 @@ export namespace MockBuilder {
...
+    export function getOtherDigestItemMock(): MockResult<DigestItem> {
+        const otherDigestValue = ByteArray.fromU8a([12, 1, 1, 1]);
+        return new MockResult(new Other(otherDigestValue), MockConstants.OTHER_DIGEST);
+    }
+
+    /**
+     * Returns SCALE Encoded ChangeTrieRoot and Instance of that Digest Item
+     */
+    export function getChangeTrieRootDigestItemMock(): MockResult<DigestItem> {
+        const changeTrieRoot = Hash.fromU8a([255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255, 255]);
```
Comment (dastanbeksamatov) [inline runtime/assembly/__tests__/mock-builder.ts:92]: Maybe create a utility function to create Hash populated with a number (for example here, it's 255)?

File: runtime/assembly/__tests__/mock-builder.ts [hunk 2]
```diff
@@ -12,85 +14,143 @@ export namespace MockBuilder {
...
         return new Header(hash69, blockNumber, hash255, hash255, digest);
     }
 
-    export function _getEmptyInherentInstance(): Inherent {
+    /**
+     * Returns a Header instance with a populated parent hash, block number, stateRoot, extrinsics root and digests.
+     * Used Internally in the mock builder
+     */
+    export function getHeaderInstanceWithDigests(): Header {
+        const hash69 = Hash.fromU8a([69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69, 69]);
```
Comment (dastanbeksamatov) [inline runtime/assembly/__tests__/mock-builder.ts:145]: Utility function would take here 69 and return Hash populated with 32 69s

File: runtime/assembly/models/header.ts [hunk 3]
```diff
@@ -30,21 +31,34 @@ export class Header {
...
     * SCALE Encodes the Header into u8[]
     */
+    // TODO fix in another PR
     toU8a(): u8[] {
-        let digest = this.digest.isSome() ? (<Hash>this.digest.unwrap()).toU8a() : Constants.EMPTY_BYTE_ARRAY;
+        let digest:u8[] = [];
+        if (this.digests.isSome()) {
+            let digestItemArray = <DigestItem[]>this.digests.unwrap();
+            const length = new CompactInt(digestItemArray.length);
+            digest.concat(length.toU8a());
```
Comment (dastanbeksamatov) [inline runtime/assembly/models/header.ts:53]: isn't it supposed to be `digest = digest.concat(length.toU8a());`

Comment (Daniel-K-Ivanov) [reply runtime/assembly/models/header.ts:53]: yes, it is. I've found the bug and fixed it in the next PR.

Good catch.

**Pull request merged**